alternator/executor.cc

```diff
@@ -3816,6 +3816,17 @@ static future<executor::request_return_type> do_query(service::storage_proxy& pr
     auto selection = cql3::selection::selection::wildcard(schema);
     query::partition_slice::option_set opts = selection->get_query_options();
     opts.add(custom_opts);
+
+    // Reverse the schema and the clustering bounds as the underlying code expects
+    // reversed queries in the native reversed format.
+    const bool reversed = opts.contains<query::partition_slice::option::reversed>();
```

Comment:
isn't that the legacy option?
how does that work during upgrade?

Reply:
No. This option is to distinguish between forward and reversed partition slices. It is set to true no matter of the format, i.e. legacy or native. Now, we always require the reversed queries to be in the native reverse format. 

Storage proxy is responsible for translating queries from native format into legacy one in case of mixed-node clusters. There is a feature flag `NATIVE_REVERSE_QUERIES` used for this purpose.